library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/SwitchCompatMapper.kt

```diff
@@ -0,0 +1,185 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.mapper
+
+import android.graphics.Rect
+import androidx.appcompat.widget.SwitchCompat
+import com.datadog.android.sessionreplay.internal.recorder.SystemInformation
+import com.datadog.android.sessionreplay.internal.recorder.ViewUtils
+import com.datadog.android.sessionreplay.internal.recorder.densityNormalized
+import com.datadog.android.sessionreplay.internal.utils.StringUtils
+import com.datadog.android.sessionreplay.model.MobileSegment
+
+internal open class SwitchCompatMapper(
+    private val textWireframeMapper: TextWireframeMapper,
+    private val stringUtils: StringUtils = StringUtils,
+    private val uniqueIdentifierGenerator: UniqueIdentifierResolver = UniqueIdentifierResolver,
+    viewUtils: ViewUtils = ViewUtils()
+) : CheckableWireframeMapper<SwitchCompat>(viewUtils) {
+
+    // region CheckableWireframeMapper
+
+    override fun resolveMainWireframes(
+        view: SwitchCompat,
+        systemInformation: SystemInformation
+    ): List<MobileSegment.Wireframe> {
+        return textWireframeMapper.map(view, systemInformation)
+    }
+
+    @Suppress("ReturnCount")
+    override fun resolveCheckedCheckable(
+        view: SwitchCompat,
+        systemInformation: SystemInformation
+    ): List<MobileSegment.Wireframe>? {
+        val thumbId = uniqueIdentifierGenerator.resolveChildUniqueIdentifier(view, THUMB_KEY_NAME)
+        val trackId = uniqueIdentifierGenerator.resolveChildUniqueIdentifier(view, TRACK_KEY_NAME)
+        val trackThumbDimensions = resolveThumbAndTrackDimensions(view, systemInformation)
```

Comment:
normally it should be:
```
 val thumbId = uniqueIdentifierGenerator.resolveChildUniqueIdentifier(view, THUMB_KEY_NAME) ?: return null
        val trackId = uniqueIdentifierGenerator.resolveChildUniqueIdentifier(view, TRACK_KEY_NAME) ?: return null
        val trackThumbDimensions = resolveThumbAndTrackDimensions(view, systemInformation) ?: return null
  ```
but the new detekt rules force me to this approach as they only allow 2 return statements / method. Especially in situations like this one I would like to override this rule if possible. @reviewers what do you think ?